--- a/src/config/WallpaperMatcher.cpp	2026-01-03 10:59:04.563049685 -0500
+++ b/src/config/WallpaperMatcher.cpp	2026-01-03 10:59:17.826651381 -0500
@@ -16,7 +16,7 @@
     }
 
     std::erase_if(m_settings, [&s](const auto& e) { return std::ranges::any_of(s, [&e](const auto& el) { return el.monitor == e.monitor; }); });
-    m_settings.append_range(std::move(s));
+    m_settings.insert(m_settings.end(), std::make_move_iterator(s.begin()), std::make_move_iterator(s.end()));
     recalcStates();
 }
 
